types/node/zlib.d.ts

```diff
@@ -214,6 +245,13 @@ declare module "zlib" {
     function createUnzip(options?: ZlibOptions): Unzip;
     type InputType = string | ArrayBuffer | NodeJS.ArrayBufferView;
     type CompressCallback = (error: Error | null, result: Buffer) => void;
+    type CompressDeflateInfoCallback = (error: Error | null, result: CompressDeflateInfoResult) => void;
+    type CompressDeflateRawInfoCallback = (error: Error | null, result: CompressDeflateInfoResult) => void;
+    type CompressGzipInfoCallback = (error: Error | null, result: CompressGzipInfoResult) => void;
+    type CompressGunzipInfoCallback = (error: Error | null, result: CompressGunzipInfoResult) => void;
+    type CompressInflateInfoCallback = (error: Error | null, result: CompressInflateInfoResult) => void;    
+    type CompressInflateRawInfoCallback = (error: Error | null, result: CompressInflateInfoResult) => void;
+    type CompressUnzipInfoCallback = (error: Error | null, result: CompressUnzipInfoResult) => void;
```

Comment:
How about to create one type generic instead of 7 new types?
```ts
type CompressInfoCallback<Engine> = (error: Error | null, result: CompressInfoResult<Engine>) => void;
 ```